controllers/onload_controller.go

```diff
@@ -209,41 +220,73 @@ func (r *OnloadReconciler) deleteLabels(ctx context.Context, namespacedName type
 
 func (r *OnloadReconciler) addKmmLabelsToNodes(ctx context.Context, onload *onloadv1alpha1.Onload) (*ctrl.Result, error) {
 	log := log.FromContext(ctx)
-	kmmPodLabelSet := labels.Set{"kmm.node.kubernetes.io/module.name": onload.Name + onloadModuleNameSuffix}
 
-	labels := labels.FormatLabels(onload.Spec.Selector)
-	labelKey := kmmLabelName(onload.Name, onload.Namespace)
-
-	nodes, err := r.listNodesWithLabels(ctx, labels)
+	// Figure out a list of nodes that match Onload's selector.
+	onloadLabels := labels.FormatLabels(onload.Spec.Selector)
+	nodes, err := r.listNodesWithLabels(ctx, onloadLabels)
 	if err != nil {
 		return nil, err
 	}
 
 	changesMade := false
 
+	addKmmLabelsToNode := func(node corev1.Node, labelKey string, podLabelSet labels.Set,
+	) (*ctrl.Result, error) {
```

Comment:
You can remove `*ctrl.Result` from the return types as you never return a non-nil value. Also, could you return whether or not the function made a change, so that you are modifying a variable defined outside of the function (I know that this version works, its just a bit ugly).

The return type could be something like `(bool, error)`

Reply:
> You can remove *ctrl.Result from the return types as you never return a non-nil value. 

It doesn't sound right because the nested function requests reconciliation if there are running Onloaded pods ("Lingering Module Pod(s)").

> its just a bit ugly

This ugliness is somewhat expected here because the function itself is non-trivial and tries to batch the reconciliations if there are any changes. (I think about the boolean as a deferred requeue/reconciliation request.) If anyone trips over this boolean, they will notice the batching quicker, what do you reckon?

If you still don't like it, maybe we should address the root cause of it and requeue without batching?

Reply:
Fair enough, leave as is then